azurerm/internal/services/kusto/kusto_cluster_resource.go

```diff
@@ -322,6 +355,44 @@ func validateAzureRMKustoClusterName(v interface{}, k string) (warnings []string
 	return warnings, errors
 }
 
+func expandOptimizedAutoScale(d *schema.ResourceData) *kusto.OptimizedAutoscale {
```

Comment:
rather than passing in `resourceData` here can we pass in the object we expect instead:

```suggestion
func expandOptimizedAutoScale(input []interface{}) *kusto.OptimizedAutoscale {
	if len(input) == 0 {
		return nil
	}
```